src/v/cluster/rm_stm.cc

```diff
@@ -1063,6 +1063,16 @@ ss::future<result<raft::replicate_result>> rm_stm::replicate_seq(
     }
     auto synced_term = _insync_term;
 
+    if (bid.first_seq > bid.last_seq) {
```

Comment:
can you add a sentence to the commit message why the validation is stronger in this case ?